macros/tables/sat.sql

```diff
@@ -11,41 +11,45 @@
 {{ dbtvault.prepend_generated_by() }}
 
 WITH source_data AS (
-    SELECT *
-    FROM {{ ref(source_model) }}
+    {%- if model.config.materialized == 'vault_insert_by_rank' %}
+    SELECT {{ dbtvault.prefix(source_cols_with_rank, 'a', alias_target='source') }}
+    {%- else %}
+    SELECT {{ dbtvault.prefix(source_cols, 'a', alias_target='source') }}
+    {%- endif %}
+    FROM {{ ref(source_model) }} AS a
+    WHERE {{ dbtvault.multikey(src_pk, prefix='a', condition='IS NOT NULL') }}
     {%- if model.config.materialized == 'vault_insert_by_period' %}
     WHERE __PERIOD_FILTER__
     {% endif %}
 ),
 
 {% if dbtvault.is_vault_insert_by_period() or is_incremental() -%}
 
-latest_sat_records_in_source AS (
+latest_records AS (
+
     SELECT {{ dbtvault.prefix(source_cols, 'a', alias_target='target') }}
-    FROM {{ this }} as a
-    JOIN source_data as b
-    ON a.{{ src_pk }} = b.{{ src_pk }}
-    WHERE 1 = 1
-    QUALIFY
-        1 = RANK() OVER (
-            PARTITION BY {{ dbtvault.prefix([src_pk], 'a') }}
-            ORDER BY {{ dbtvault.prefix([src_ldts], 'a') }} DESC
-        )
+    FROM (
```

Comment:
Feels like we can get rid of this extra nesting with a `WHERE 1 = 1 QUALIFY ...` clause